src/main/java/com/google/finapp/SpannerDaoImpl.java

```diff
@@ -109,6 +120,10 @@ public void moveAccountBalance(ByteArray fromAccountId, ByteArray toAccountId, B
                 ImmutableMap<ByteArray, BigDecimal> accountBalances =
                     readAccountBalances(fromAccountId, toAccountId, transaction);
 
+                if (accountBalances.get(fromAccountId).subtract(amount).signum() == -1) {
+                  throw new IllegalArgumentException("Account balance cannot be negative");
```

Comment:
Similar comment, here it would be good to print both the starting amount and the amount that we tried to move.